src/launchpad/analyzers/apple.py

```diff
@@ -120,12 +124,28 @@ def analyze(self, artifact: AppleArtifact) -> AppleAnalysisResults:
             )
             treemap = treemap_builder.build_file_treemap(file_analysis)
 
+        insights = {}
+        if not self.skip_insights:
+            logger.info("Generating insights from analysis results")
+            temp_results = AppleAnalysisResults(
+                app_info=app_info,
+                file_analysis=file_analysis,
+                binary_analysis=binary_analysis,
+                analysis_duration=0,
+                treemap=treemap,
+                insights=InsightResults(),
+            )
+            insights = {
+                "duplicate_files": self.INSIGHTS[0].generate_insight(temp_results),
```

Comment:
This part is weird but i'll clean it up when I add in some more insights.

Reply:
Yeah generally I'm a bit confused here why this block is taking place to generate the temp results, this whole block is a bit hard to follow.